llvm/lib/SYCLLowerIR/SYCLPropagateAspectsUsage.cpp

```diff
@@ -665,6 +665,24 @@ void processDeclaredVirtualFunctionSets(
     StringMap<SmallVector<Function *, 4>> &VirtualFunctionSets) {
   if (!F->hasFnAttribute("calls-indirectly"))
     return;
+
+  // "Construction" kernels which reference vtables but do not actually
+  // perform any virtual calls have the calls-indirectly attribute
+  // attached by SYCLVirtualFunctionAnalysis pass. We do not want to
+  // attach sycl_used_aspects metadata to such kernels.
+  bool hasVirtualCall = false;
+  for (const Instruction &I : instructions(F)) {
```

Comment:
This is purely a FYI, LLVM has `llvm::none_of` which works with their ranges if you want to rewrite this into `if (none_of(instructions(F), []() { multi-line lambda here }) return;`. See [STLExtras](https://llvm.org/doxygen/STLExtras_8h.html)